src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure.NotebookTests/Utils.cs

```diff
@@ -0,0 +1,134 @@
+using GC.Infrastructure.NotebookTests.NotebookParser;
+using Newtonsoft.Json;
+using System.Diagnostics;
+
+namespace GC.Infrastructure.NotebookTests
+{
+    internal static class Utils
+    {
+        internal const int TIMEOUT = 120_000;
+        public static bool CheckDotnetReplInstalled()
+        {
+            using (Process process = new Process())
+            {
+                process.StartInfo.FileName = "dotnet-repl";
+                process.StartInfo.Arguments = "--version";
+                process.StartInfo.UseShellExecute = false;
+
+                try
+                {
+                    process.Start();
+                    process.WaitForExit();
+                }
+
+                catch (System.ComponentModel.Win32Exception)
+                {
+                    return false;
+                }
+
+                return process.ExitCode == 0;
+            }
+        }
+
+        public static void InstallDotnetRepl()
+        {
+            using (Process process = new Process())
+            {
+                process.StartInfo.FileName = "dotnet";
+                process.StartInfo.Arguments = "tool install --local dotnet-repl";
+                process.StartInfo.UseShellExecute = false;
+                process.Start();
+                process.WaitForExit();
+            }
+        }
+
+        public static Process SetupNotebookProcessRun(string notebookPath, string? outputNotebookPath = null)
+        {
+            Process process = new Process();
+            process.StartInfo.FileName = "dotnet-repl";
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.Arguments = $"--run {notebookPath} --exit-after-run";
+
+            // Optionally include the output notebook path in case we care about the output.
+            if (!string.IsNullOrEmpty(outputNotebookPath))
+            {
+                process.StartInfo.Arguments += $" --output-path {outputNotebookPath}";
+            }
+
+            // The working directory should be the directory that the notebook is in.
+            process.StartInfo.WorkingDirectory = Path.GetDirectoryName(notebookPath);
+            return process;
+        }
+
+        public static void RunNotebookThatsExpectedToPass(string notebookPath) 
+        {
+            string tempPathForOutputNotebook = Path.GetTempFileName();
+            using (Process dotnetReplProcess = SetupNotebookProcessRun(notebookPath, tempPathForOutputNotebook))
+            {
+                dotnetReplProcess.Start();
+                dotnetReplProcess.WaitForExit(TIMEOUT);
+
+                // If notebook execution fails, parse the notebook and find the error.
+                if (dotnetReplProcess.ExitCode != 0)
+                {
+                    ParseNotebookAndFindErrors(tempPathForOutputNotebook, notebookPath);
+                }
+
+                else
+                {
+                    dotnetReplProcess.HasExited.Should().BeTrue();
```

Comment:
I don't think this does anything.  If the process hasn't exited, `ExitCode` will throw.  I think you want to assert that the return value of `WaitForExit` is `true`.